packages/cli/yaml/validator/src/rules/valid-pagination/validateOffsetPagination.ts

```diff
@@ -124,3 +134,40 @@ function isValidOffsetType({ resolvedType }: { resolvedType: ResolvedType | unde
     }
     return primitiveType === "INTEGER" || primitiveType === "LONG" || primitiveType === "DOUBLE";
 }
+
+function validateHasNextPageProperty({
+    endpointId,
+    resolvedResponseType,
+    typeResolver,
+    file,
+    offsetPagination
+}: {
+    endpointId: string;
+    resolvedResponseType: ResolvedType;
+    typeResolver: TypeResolver;
+    file: FernFileContext;
+    offsetPagination: RawSchemas.OffsetPaginationSchema;
+}): RuleViolation[] {
+    if (offsetPagination["has-next-page"] == null) {
+        return [];
+    }
+    return validateResponseProperty({
+        endpointId,
+        typeResolver,
+        file,
+        responseProperty: offsetPagination["has-next-page"],
+        resolvedResponseType,
+        propertyValidator: {
+            propertyID: "has-next-page",
+            validate: isValidHasNextPageType
+        }
+    });
+}
+
+function isValidHasNextPageType({ resolvedType }: { resolvedType: ResolvedType | undefined }): boolean {
+    const primitiveType = maybePrimitiveType(resolvedType);
+    if (primitiveType == null) {
+        return false;
+    }
+    return primitiveType === "BOOLEAN";
```

Comment:
I could see us eventually supporting `string` so that it can be used similar to an empty cursor token, but let's start with this for now, especially because there aren't any IR changes required.